### PR TITLE
Add CRUD (create,delete) tests for console extension resources

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Since we are loading these resources from generated YAML in `/vendor`, lets make sure we can `CREATE`/`DELETE` one of each.

/assign @spadgett @jhadvig

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
/retest

`level=fatal msg="Terraform destroy: failed to destroy using Terraform"`

File: test/e2e/extensions_crud_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+			Links: []consolev1.CLIDownloadLink{{
+				Text: "download test",
+				Href: "https://example.com",
+			}},
+		},
+	}
+
+	download, err := client.ConsoleCliDownloads.Create(download)
+	if err != nil {
+		t.Fatalf("%v", err)
```
Comment (jhadvig) [inline test/e2e/extensions_crud_test.go]: nit: let's make this one more noisy:
```suggestion
		t.Fatalf("could not create ConsoleCliDownloads custom resource: %v", err)
```

File: test/e2e/extensions_crud_test.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	}
+	err = client.ConsoleCliDownloads.Delete(download.Name, &v1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("%v", err)
```
Comment (jhadvig) [inline test/e2e/extensions_crud_test.go]: nit: let's make this one more noisy:
```suggestion
		t.Fatalf("could not delete ConsoleCliDownloads custom resource: %v", err)
```

**Comment by benjaminapetersen** [comment]
@jhadvig udpated, increased noise for all tests!

**Pull request merged**